src/routes/docs/products/functions/functions/+page.markdoc

```diff
@@ -0,0 +1,185 @@
+![Create project screen](/images/docs/functions/dark/template.png)
+{% /only_dark %}
+{% only_light %}
+![Create project screen](/images/docs/functions/template.png)
+{% /only_light %}
+{% /tabsitem %}
+
+{% tabsitem #manual title="Manual" %}
+You can also create Appwrite Functions manually by uploading your code in a zipped file.
+In the **Create Function** modal, click **create a function manually** at the bottom to switch to manual create wizard.
+You will asked to upload a zip file with your code.
+
+First, navigate inside the folder that contains your dependency file. 
+For example, when you list the content of your folder for a Node.js function, it will look like this.
+
+```text
+.
+├── package.json
+└── index.js
+```
+
+Package your code files into the `.tar.gz` format. **Don't include your dependencies folder**, such as **node_modules**.
+
+{% multicode %}
+  ```bash
+  tar --exclude code.tar.gz -czf code.tar.gz .
+  ```
+  ```cmd
+  tar --exclude code.tar.gz -czf code.tar.gz .
+  ```
+  ```powershell
+  tar --exclude code.tar.gz -czf code.tar.gz .
+  ```
+{% /multicode %}
+
+Upload your `.tar.gz` file and specify the entry point of your function, in this case `index.js`.
+Remember to specify the build commands for your function to install dependencies.
+{% /tabsitem %}
+
+{% tabsitem #cli title="CLI" %}
+You can create functions using the CLI, without needing to access the console.
+Before you can deploy with the Appwrite CLI, make sure you've [initialized the CLI.](/docs/tooling/command-line/installation)
+
+To deploy with the Appwrite CLI, your function must be added to your `appwrite.json` config file. 
+Use the `appwrite init function` method to create a starter function, then paste in your function code.
+
+```bash
+appwrite init function
+```
+
+The init command will initialize a folder with a starter function code.
+To deploy the generated code, add dependencies, and deploy the function using the following command:
+
+```bash
+appwrite deploy function
+```
+{% /tabsitem %}
+{% /tabs %}
+
+After creating your function, create a [deployment](/docs/products/functions/deployments) to add code to the function.
+
+# Configuration {% #configuration %}
+
+## Name {% #name %}
+You can update the name of your function by navigating to your function > **Settings** > **Name**.
+Update your function's name and click **Update**.
+Functions are executed using it's ID, updating name does not affect references to your function.
+
+## Runtime {% #runtime %}
+Every deployment of a function uses the same runtime. 
+You can update the runtime of a function by navigating to function > **Settings** > **Runtime**.
+Select a new runtime and click **Update**.
+
+{% info title="Redeployment required" %}
+This change requires your function to be redeployed to take effect.
+{% /info %}
+
+When you update a function's runtime, the code of your function may not be compatible with the new runtime.
+Remember to update your code and redeploy your function.
```

Comment:
```suggestion
When you update a function's runtime, you may need to update code of your function to make it compatible with the new runtime.
```

Above we already have big alert box teiing to redeploy.

Reply:
Done